pkg/controller/gitopsconfig/controller.go

```diff
@@ -417,3 +437,191 @@ func isOwner(owner, owned metav1.Object) bool {
+	if active > 0 {
+		return "INPROGRESS"
+	}
+	if succeeded != 0 {
+		return "SUCCESS"
+	}
+	if failed != 0 {
+		return "FAILED"
+	}
+	return "INPROGRESS"
+
+}
+
+func (job *jobdata) checkResult() (jobState gitopsv1alpha1.GitOpsConfigStatus, err error) {
+	var jobOutput batchv1.Job
+	err = job.Reconcile.client.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      job.Name,
+			Namespace: job.Namespace,
+		},
+		&jobOutput)
+
+	if err != nil {
+		log.Error(err, "Error in GetJob for CheckResult")
+		return jobState, err
+	}
+
+	jobState.State = getJobStatus(jobOutput.Status.Succeeded, jobOutput.Status.Failed, jobOutput.Status.Active)
+	if jobOutput.Status.Active > 0 && jobOutput.Status.Failed > 0 {
+		jobState.Message = "job failed, retrying..."
+	}
+	jobState.StartTime = jobOutput.Status.StartTime
+	jobState.CompletionTime = jobOutput.Status.CompletionTime
+
+	return jobState, nil
+}
+
+func (job *jobdata) watchjob(instanceName, instanceNamespace string) {
+	log.Info("Watching the Job: " + job.Name)
+	var initialized bool
+	for {
+		if !job.NextSchedule.IsZero() && time.Now().After(job.NextSchedule) {
+			log.Info("Breaking out of infinite loop")
+			return
+		}
+		time.Sleep(10 * time.Second)
+		jobState, err := job.checkResult()
+		if err != nil {
+			continue
+		}
+
+		if jobState.State != "INPROGRESS" {
+			err := job.updateStatus(instanceName, instanceNamespace, jobState)
+			if err != nil {
+				log.Error(err, "Error in Updating the job status")
+			}
+			return
+		}
+
+		if !initialized {
+			initialized = true
+			log.Info("Initial Update")
+			err := job.updateStatus(instanceName, instanceNamespace, jobState)
+			if err != nil {
+				log.Error(err, "Error in Initial Update of job Status")
+			}
+		}
+
+	}
+}
+
+//Pass the instanceName and instanceNamespace from the instance and not from job
+func (job *jobdata) updateStatus(instanceName string, instanceNamespace string, jobState gitopsv1alpha1.GitOpsConfigStatus) error {
+	var newinstance = new(gitopsv1alpha1.GitOpsConfig)
+
+	err := job.Reconcile.client.Get(context.TODO(), types.NamespacedName{Name: instanceName, Namespace: instanceNamespace}, newinstance)
+	if err != nil {
+		log.Error(err, "Error in gettting the instance")
+		return err
+	}
+	newinstance.Status = jobState
+	err = job.Reconcile.client.Status().Update(context.TODO(), newinstance)
+	if err != nil {
+		log.Error(err, "Failed to update status", "job", job.Name)
+		return err
+	}
+
+	log.Info("Status updated as " + jobState.State + " for job : " + job.Name)
+	return nil
+}
+
+func (job *jobdata) getCronJobData() (batchv1beta1.CronJob, error) {
+	var result batchv1beta1.CronJob
+	err := job.Reconcile.client.Get(context.TODO(),
+		types.NamespacedName{Name: job.Name, Namespace: job.Namespace},
+		&result)
+	return result, err
+}
+
+func watchJobForStatus(r *Reconciler, jobName string, namespace string, instanceName string, instanceNamespace string, nextSchedule time.Time) {
+	if runningJobMap[jobName] {
```

Comment:
as we discussed on chat, I believe we should not do this here; instead, we should rather fix #179 as a completely separate effort. Please remove all usage of runningJobMap in watchJobForStatus.